products/BellHybrid/apps/application-bell-meditation-timer/widgets/SummaryListItem.cpp

```diff
@@ -1,46 +1,67 @@
 
 namespace app::meditation
 {
     using namespace gui;
-    SummaryListItem::SummaryListItem(const std::string &titleText, const std::string &valueText)
+    SummaryListItem::SummaryListItem(const std::string &topDescription,
+                                     const std::chrono::minutes total,
+                                     const std::chrono::minutes average)
+        : BellSideListItemWithCallbacks(topDescription)
     {
-        setMinimumSize(width, height);
-        setMargins(Margins(0, style::margins::big, 0, style::margins::huge));
-        activeItem = false;
-
-        body = new VBox(this, 0, 0, 0, 0);
-        body->setEdges(RectangleEdge::None);
-
-        title = new Text(body, 0, 0, 0, 0);
-        title->setMinimumSize(width, title_height);
-        title->setFont(style::window::font::bigbold);
-        title->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Top));
-        title->setText(titleText);
-
-        value = new Text(body, 0, 0, 0, 0);
-        value->setMinimumSize(width, value_height);
-        value->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Bottom));
-        value->setFont(style::window::font::big);
-        value->setText(valueText);
-
-        dimensionChangedCallback = [&]([[maybe_unused]] Item &item, const BoundingBox &newDim) -> bool {
-            body->setArea({0, 0, newDim.w, newDim.h});
-            return true;
-        };
+
+        auto duoBox = new VBox(body->getCenterBox());
+        duoBox->setEdges(RectangleEdge::None);
+        duoBox->setAlignment(Alignment(gui::Alignment::Horizontal::Center, gui::Alignment::Vertical::Center));
+        duoBox->setMaximumSize(::style::bell_base_layout::w, ::style::bell_base_layout::h);
+
+        auto centerText = new TextFixedSize(duoBox);
+        centerText->setMinimumSize(style::bell_base_layout::w, 2 * (style::bell_base_layout::center_layout_h / 3));
+        centerText->setFont(style::bell_sidelist_item::title_font);
+        centerText->setEdges(RectangleEdge::None);
+        centerText->activeItem = false;
+        centerText->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Top));
+        centerText->setRichText(utils::translate("app_meditation_summary_total"),
+                                text::RichTextParser::TokenMap({{"$VALUE", format(total)}}));
+        centerText->drawUnderline(false);
+
+        auto centerText2 = new TextFixedSize(duoBox);
+        centerText2->setFont(style::bell_sidelist_item::title_font);
+        centerText2->setMinimumSize(style::bell_base_layout::w, style::bell_base_layout::center_layout_h / 3);
+        centerText2->setEdges(RectangleEdge::None);
+        centerText2->activeItem = false;
+        centerText2->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Bottom));
+        centerText2->setRichText(utils::translate("app_meditation_summary_average"));
+        centerText2->drawUnderline(false);
```

Comment:
add private function to call repeated lines

auto centerText = new TextFixedSize(duoBox);
auto centerText2 = new TextFixedSize(duoBox);

Reply:
I can add such a method to separate the common code, but IMHO it would not give us any benefits. These elements are not the same mostly, and as a result, they need different parameters to set.